### PR TITLE
Attack Logic Modification

### PR DESCRIPTION
**Pull request by saeidalz13**
# Overview

As per our discussion, attack handler was modified.

- `RespAttack` is unified for both players, modified with new fields.
- Both defender and attacker will receive `RespAttack` in case of a valid attack.

### CODE REVIEW

File: api/handlers_api.go [hunk 1]
```diff
@@ -161,49 +166,50 @@ func (w *Request) HandleAttack() (*md.Message[md.RespAttack], *md.Player) {
...
-	if positionCode == md.PositionStateDefenceGridEmpty {
+	if positionCode == md.PositionStateAttackGridMiss {
 		resp.AddPayload(md.RespAttack{
-			PositionState: md.PositionStateAttackGridMiss,
-			IsTurn:        attacker.IsTurn,
+			X:               x,
+			Y:               y,
+			PositionState:   md.PositionStateAttackGridMiss,
+			SunkenShipsHost: game.HostPlayer.SunkenShips,
+			SunkenShipsJoin: game.JoinPlayer.SunkenShips,
```
Comment (mori-ahk) [inline api/handlers_api.go:187]: Am I blind or you forgot to pass `IsTurn` in the `RespAttack` ?

File: api/ws_api.go [hunk 2]
```diff
@@ -189,18 +189,27 @@ func (s *Server) manageWsConn(ws *websocket.Conn) {
...
 				continue
 			}
 
+			// defender turn is set to true
+			resp.Payload.IsTurn = true 
+			if err := defender.WsConn.WriteJSON(resp); err != nil {
+				log.Println(err)
+				continue
+			}
+
```
Comment (mori-ahk) [inline api/ws_api.go:208]: can we use `SendMsgToBothPlayers` function here?

File: api/handlers_api.go (hunk 1 above)
Comment (mori-ahk) [reply api/handlers_api.go:187]: Nvm I saw what you did there

**Review by mori-ahk — APPROVED** [review]
Apart from a comment, the rest LGTM ✅

File: api/handlers_api.go (hunk 1 above)
Comment (saeidalz13) [reply api/handlers_api.go:187]: Lool, yea I put it down there to reflect the final status of sunkenships

File: api/ws_api.go (hunk 2 above)
Comment (saeidalz13) [reply api/ws_api.go:208]: I thought about it. In that case we need to return game too... I mean we can do that I don't know.

Comment (saeidalz13) [reply api/ws_api.go:208]: I'm writing tests for the attack. I'll apply this change (`SendMsgToBothPlayers`) in that PR.

**Pull request merged**